README.md

```diff
@@ -99,27 +99,90 @@ both at a short term but also at a long term level. The concept of "Memory" exis
 
 ## 🤖 Developer Guide
 
-To begin developing on this project, first clone to the repo locally.
-To install requirements, run `pip install -r requirements.txt`.
-This will install all requirements for running the package, examples, linting, formatting, and tests.
+To begin developing on this project, first clone the repo locally.
+
+### Quick Start
+
+This project uses [Poetry](https://python-poetry.org/) as a dependency manager. Check out Poetry's own [documentation on how to install it](https://python-poetry.org/docs/#installation) on your system before proceeding.
+
+To install requirements:
+
+```bash
+poetry install -E all
+```
+
+This will install all requirements for running the package, examples, linting, formatting, and tests. Note the `-E all` flag will install all optional dependencies necessary for integration testing.
+
+Now, you should be able to run the common tasks in the following section.
+
+### Common Tasks
+
+#### Code Formatting
 
 Formatting for this project is a combination of [Black](https://black.readthedocs.io/en/stable/) and [isort](https://pycqa.github.io/isort/).
-To run formatting for this project, run `make format`.
+
+To run formatting for this project:
+
+```bash
+make format
+```
+
+#### Linting
 
 Linting for this project is a combination of [Black](https://black.readthedocs.io/en/stable/), [isort](https://pycqa.github.io/isort/), [flake8](https://flake8.pycqa.org/en/latest/), and [mypy](http://mypy-lang.org/).
-To run linting for this project, run `make lint`.
+
+To run linting for this project:
+
+```bash
+make lint
+```
+
 We recognize linting can be annoying - if you do not want to do it, please contact a project maintainer and they can help you with it. We do not want this to be a blocker for good code getting contributed.
 
+#### Testing
+
 Unit tests cover modular logic that does not require calls to outside apis.
-To run unit tests, run `make tests`.
+
+To run unit tests:
+
+```bash
+make tests
+```
+
 If you add new logic, please add a unit test.
 
 Integration tests cover logic that requires making calls to outside APIs (often integration with other services).
-To run integration tests, run `make integration_tests`.
+
+To run integration tests:
+
+```bash
+make integration_tests
```

Comment:
Integration tests are failing locally for me, so there might be more documentation necessary to get this working. I'm seeing the same behavior on mainline, so it's not due to the Poetry integration.

```
=================================== short test summary info =====================================
FAILED tests/integration_tests/test_nlp_text_splitters.py::test_nltk_text_splitter - LookupError:
FAILED tests/integration_tests/test_nlp_text_splitters.py::test_spacy_text_splitter - OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or ...
FAILED tests/integration_tests/test_serpapi.py::test_call - pydantic.error_wrappers.ValidationError: 1 validation error for SerpAPIWrapper
FAILED tests/integration_tests/chains/test_pal.py::test_math_prompt - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/chains/test_pal.py::test_colored_object_prompt - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/chains/test_react.py::test_react - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/chains/test_self_ask_with_search.py::test_self_ask_with_search - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/chains/test_sql_database.py::test_sql_database_run - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/chains/test_sql_database.py::test_sql_database_run_update - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/embeddings/test_cohere.py::test_cohere_embedding_documents - pydantic.error_wrappers.ValidationError: 1 validation error for CohereEmbeddings
FAILED tests/integration_tests/embeddings/test_cohere.py::test_cohere_embedding_query - pydantic.error_wrappers.ValidationError: 1 validation error for CohereEmbeddings
FAILED tests/integration_tests/embeddings/test_huggingface_hub.py::test_huggingfacehub_embedding_documents - pydantic.error_wrappers.ValidationError: 1 validation error for HuggingFaceHubEmbeddings
FAILED tests/integration_tests/embeddings/test_huggingface_hub.py::test_huggingfacehub_embedding_query - pydantic.error_wrappers.ValidationError: 1 validation error for HuggingFaceHubEmbeddings
FAILED tests/integration_tests/embeddings/test_openai.py::test_openai_embedding_documents - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAIEmbeddings
FAILED tests/integration_tests/embeddings/test_openai.py::test_openai_embedding_query - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAIEmbeddings
FAILED tests/integration_tests/llms/test_ai21.py::test_ai21_call - pydantic.error_wrappers.ValidationError: 1 validation error for AI21
FAILED tests/integration_tests/llms/test_cohere.py::test_cohere_call - pydantic.error_wrappers.ValidationError: 1 validation error for Cohere
FAILED tests/integration_tests/llms/test_huggingface_hub.py::test_huggingface_text_generation - pydantic.error_wrappers.ValidationError: 1 validation error for HuggingFaceHub
FAILED tests/integration_tests/llms/test_huggingface_hub.py::test_huggingface_text2text_generation - pydantic.error_wrappers.ValidationError: 1 validation error for HuggingFaceHub
FAILED tests/integration_tests/llms/test_huggingface_hub.py::test_huggingface_call_error - pydantic.error_wrappers.ValidationError: 1 validation error for HuggingFaceHub
FAILED tests/integration_tests/llms/test_manifest.py::test_manifest_wrapper - ValueError: OpenAI API key not set. Set OPENAI_API_KEY environment variable or pass through `...
FAILED tests/integration_tests/llms/test_nlpcloud.py::test_nlpcloud_call - pydantic.error_wrappers.ValidationError: 1 validation error for NLPCloud
FAILED tests/integration_tests/llms/test_openai.py::test_openai_call - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/llms/test_openai.py::test_openai_extra_kwargs - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/llms/test_openai.py::test_openai_stop_valid - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/llms/test_openai.py::test_openai_stop_error - pydantic.error_wrappers.ValidationError: 1 validation error for OpenAI
FAILED tests/integration_tests/vectorstores/test_elasticsearch.py::test_elasticsearch - elastic_transport.ConnectionError: Connection error caused by: ConnectionError(Connection err...
FAILED tests/integration_tests/vectorstores/test_elasticsearch.py::test_elasticsearch_with_metadatas - elastic_transport.ConnectionError: Connection error caused by: ConnectionError(Connection err...
FAILED tests/integration_tests/vectorstores/test_faiss.py::test_faiss_with_metadatas - AssertionError: assert {'_dict': {'1...kup_index=0)}} == {'_dict': {'0...kup_index=0)}}
```